src/contracts/core/AVSDirectory.sol

```diff
@@ -264,25 +271,38 @@ contract AVSDirectory is
         _deregisterFromOperatorSets(msg.sender, operator, operatorSetIds);
     }
 
-
     /**
-     * @notice Called by operators to set their allocation delay. 
+     * @notice Called by operators to set their allocation delay.
      * @param delay the allocation delay in seconds
      * @dev msg.sender is assumed to be the operator
      * @dev Fails if setting a delay would result in the ability to set an
-     *      an allocation that is active BEFORE the latest set allocation. 
+     *      an allocation that is active BEFORE the latest set allocation.
      */
-    function setAllocationDelay(uint32 delay) external {
+    function setAllocationDelay(
+        uint32 delay
+    ) external {
         require(
             delegation.isOperator(msg.sender),
             "avsDirectory.initializeAllocationDelay: operator not registered to EigenLayer yet"
         );
+
+        AllocationDelayInfo storage delayInfo = _allocationDelayInfo[msg.sender];
+
         require(
-            delay + block.timestamp >= latestAllocationEffectTimestamp[msg.sender],
+            delay + block.timestamp >= delayInfo.latestAllocationEffectTimestamp,
             "avsDirectory.initializeAllocationDelay: new delay must be after latestAllocationEffectTimestamp"
         );
-        
-        _operatorAllocationDelay[msg.sender] = delay;
+
+        bool pendingInEffect = delayInfo.pendingDelay != 0 && block.timestamp >= delayInfo.pendingDelayEffectTimestamp;
+
+        if (pendingInEffect) {
+            delayInfo.delay = delayInfo.pendingDelay;
+        }
+
+        delayInfo.pendingDelay = delay;
+        delayInfo.pendingDelayEffectTimestamp = uint32(block.timestamp + ALLOCATION_DELAY_CONFIGURATION_DELAY);
```

Comment:
`ALLOCATION_DELAY_CONFIGURATION_DELAY` should be `delayInfo.delay`

Reply:
or rather `operatorAllocationDelay`

Reply:
what if its the first time being set? Is it just the default deallocation delay?